### PR TITLE
chitter_challenge - Gina

### PR DESCRIPTION
**Pull request by Gina-Frankel**

### CODE REVIEW

File: playing_with_time.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+Time.now
```
Comment (hturnbull93) [inline playing_with_time.rb:1]: I guess this is a work in progress file? :thinking:

File: spec/features/viewing_messages_spec.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    expect(page).to have_content "Harry"
+    expect(page).to have_content "Guy"
+  end 
+
+  scenario 'seeing the time ' do
+    connection = PG.connect(dbname: 'chitter_test')
+    Messages.create(message: 'What is the time?', name: 'Mr Time')
+    visit ('/')
+    
+    expect(page).to have_content (/\w{4}-.{2}-\w{2}.\w{2}:\w{2}/)
```
Comment (hturnbull93) [inline spec/features/viewing_messages_spec.rb:24]: I'm not sure what this content is, a regex for the date string? :thinking:

File: spec/messages_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+require 'messages'
+
+describe Messages do 
+  context '.get_all' do 
+    connection = PG.connect(dbname: 'chitter_test')
```
Comment (hturnbull93) [inline spec/messages_spec.rb:5]: This is no longer needed right?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,6 +1,31 @@
...
+
+
+ENV['RACK_ENV'] = 'test'
+
+
+# Setting up test environment 
+ENV['ENVIRONMENT'] = 'test'
+require_relative './setup_test_database'
+
+ENV['ENVIRONMENT'] = 'test'
```
Comment (hturnbull93) [inline spec/spec_helper.rb:15]: Setting ENV twice, just to be sure? :100:

File: views/index.erb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+     alt="Grapefruit slice atop a pile of other slices">
+
+
+<main>
+    <p> Welcome to  <em>Chitter</em> !</p>
+</main>
+
+
+<dl>
+  <% @messages.reverse_each do |message|%>
```
Comment (hturnbull93) [inline views/index.erb:18]: This logic should probably be in the model, no? :thinking:

File: views/index.erb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    <dt><strong><%= message.name %></strong>  </dt>
+    <dd> <em>PEEPS</em> <%= message.message %> <%=Time.now%>  </dd>
```
Comment (hturnbull93) [inline views/index.erb:20]: Is `Time.now` a stand in for having the peep know its own time?

File: views/index.erb [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    
+  <% end %>
+</dl> 
+
+
+
+<form action "/messages" method = "post" >
+<input type="text" name="message" placeholder ="What is  happening?">
+<input type="text" name="name" placeholder ="name">
+<input type="submit" value="peep">
```
Comment (hturnbull93) [inline views/index.erb:30]: I like the peep form being on the main page, more elegant than moving to a separate page. :+1:

File: lib/messages.rb [hunk 8]
```diff
@@ -0,0 +1,37 @@
+require 'pg'
+
+class Messages
+
+  attr_reader :id, :message, :name
+
+  def initialize(id:, message:, name:) #(id:, name:, message:)
+    @id = id
+    @message = message
+    @name = name
```
Comment (hturnbull93) [inline lib/messages.rb:10]: Should a peep be responsible for knowing the authors name?
Perhaps that should be a responsibility of an Author, or User class?

File: README.md [hunk 9]
```diff
@@ -1,6 +1,104 @@
...
+  Client->>+Server: POST request
+  Server->>-Client: redirects to message page??
+  Client->>-User: new page with "messages"
+  
+	
+```
+
+
+## Wireframes
+![](mockups/2020-04-12-09-12-01.png)
```
Comment (hturnbull93) [inline README.md:96]: Nice wireframe diagram :+1:

File: app.rb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+    get '/' do 
+      @messages = Messages.get_all
+      erb(:index)
+    end 
+
+    post '/' do 
+      Messages.create(message: params[:message],name: params[:name])
+      redirect '/'
+    end
```
Comment (hturnbull93) [inline app.rb:14]: Is this route RESTful? Perhaps this should be `post '/messages'` :thinking: 
In the index.erb file there is a form that posts to /messages even, is this a WIP? :thinking:

File: lib/messages.rb [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+  def self.create(message:, name:)
+    if ENV['ENVIRONMENT'] =='test'
+      connection = PG.connect(dbname: 'chitter_test');
+    else 
+      connection = PG.connect(dbname: 'chitter');
+    end 
+    result = connection.exec("INSERT INTO peeps (message, name) VALUES('#{message}', '#{name}') RETURNING message, name, id ")
+    Messages.new(id: result[0]['id'], message: result[0]['message'], name: result[0]['name']) #, message: result[0]['name'], name: result[0]['messsage'])
+
+  end
```
Comment (hturnbull93) [inline lib/messages.rb:36]: Good solid write and read methods :+1:

File: views/index.erb [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+  <% @messages.reverse_each do |message|%>
+    <dt><strong><%= message.name %></strong>  </dt>
+    <dd> <em>PEEPS</em> <%= message.message %> <%=Time.now%>  </dd>
+    
+  <% end %>
+</dl> 
+
+
+
+<form action "/messages" method = "post" >
```
Comment (hturnbull93) [inline views/index.erb:27]: this form posts to /messages, but there appears not to be a /messages route in the controller :worried:

File: playing_with_time.rb (hunk 1 above)
Comment (Gina-Frankel) [reply playing_with_time.rb:1]: that was just a file I was using to work out how to get the time to save .. I should have deleted it

**Pull request closed without merging**